### ISSUE TITLE
Changing column order after hiding columns in Table visualization settings breaks column ordering

### ISSUE THREAD

**Issue by likeshumidity** [metabase/metabase#30480]
### Describe the bug

- Changing column order and hiding columns breaks column ordering
- In newer versions of Metabase, preview button fixes, but preview must be clicked between every time a column is hidden or the order is changed, so if the question takes 30 seconds to run, then it takes 30+ seconds each time you want to move and hide a column
- Affects both Native SQL and GUI Questions

### To Reproduce

1. Create a question from Sample Database > Accounts
2. Click ⚙️  to edit Visualization settings for the Table
3. Move Country to first position in the left panel menu
4. Hide Email in the left panel menu
5. Try to move Active Subscription to first position in the left panel menu, and see the issue, it returns to its original position

### Expected behavior

Expect hiding and reordering columns to continue to work without clicking preview between each operation

### Logs

- no errors in logs

### Information about your Metabase installation

```JSON
- Metabase version: `master` as of 2023-05-02 and 1.46.3.1 and 1.46.2
```

### Severity

P2

**Comment by paoliniluis** [comment]
Dupe of https://github.com/metabase/metabase/issues/29287 right?

**Comment by likeshumidity** [comment]
@paoliniluis yes, didn't see that one, though the behavior is slightly different in 46 than in 45. In 45, columns behave seemingly erratically, in 46, unable to change column order without clicking preview (but no message suggesting that).
